app/redux/selectors/SetsSelectors.js

```diff
@@ -259,4 +261,48 @@ export const hasChangesToSync = (state) => {
     return (root.setIDsToUpload.length > 0 || root.setIDsBeingUploaded.length > 0);
 };
 
+// collapsed metrics
+
+export const getBestAvgVelocityEver = (state, set) => {
+    let historySets = getHistorySetsChronological(state);
+
+    // find all instances of this exercise with weight and reps
+    let matchedSets = historySets.filter(historySet => historyFilter(historySet, set));
+
+    let avgVs = matchedSets.map((matchedSet) => {
+        return CollapsedMetrics.getAvgVelocities(matchedSet);
+    });
+
+    avgVs = avgVs.reduce((a, b) => a.concat(b), []);
+    
+    if (avgVs.length > 0) {
+        return Math.max(...avgVs);
+    } else {
+        return null;
+    }
+};
+
+export const getBestROMEver = (state, set) => {
+    let historySets = getHistorySetsChronological(state);
+
+    // find all instances of this exercise with weight and reps
+    let matchedSets = historySets.filter(historySet => historyFilter(historySet, set));
+
+    let roms = matchedSets.map((matchedSet) => {
+        return CollapsedMetrics.getROMs(matchedSet);
+    });
+
+    roms = roms.reduce((a, b) => a.concat(b), []);
+    
+    if (roms.length > 0) {
+        return Math.max(...roms);
+    } else {
+        return null;
+    }
+};
+
 export const getRevision = (state) => stateRoot(state).revision;
+
+function historyFilter(historySet, set) {
```

Comment:
- [ ] rename to areSetsComparable

- [ ] remove extra space after &&

Reply:
Will do at the end